### PR TITLE
Fix JREContainerProvider.getJREContainerEntry(IJavaProject) to also work with Java 11 and Java 17

### PR DESCRIPTION
**Pull request by cdietrich**
Fix JREContainerProvider.getJREContainerEntry(IJavaProject) to also work with Java 11 and Java 17

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JREContainerProvider.java [hunk 1]
```diff
@@ -105,12 +105,11 @@ protected static IPath newPreferredContainerPath() {
 	}
 
 	public static IClasspathEntry getJREContainerEntry(IJavaProject javaProject) throws JavaModelException {
-		IClasspathEntry defaultJREContainerEntry = getDefaultJREContainerEntry();
-
+		IPath defaultJREContainerPathPrefix = JavaRuntime.newDefaultJREContainerPath().append(StandardVMType.ID_STANDARD_VM_TYPE);
```
Comment (szarnekow) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JREContainerProvider.java:108]: It looks as if this should make sense, yes.

**Review by szarnekow — APPROVED** [review]
Shall we deprecate / remove the JRE15 (as in Java 1.5) stuff from this class?

**Comment by cdietrich** [comment]
there is a lot more jre 5/6 stuff. also all over xbase i guess

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JREContainerProvider.java [hunk 2]
```diff
@@ -83,6 +83,7 @@ private static IVMInstall defaultVMInstall() {
 	/**
 	 * @return JRE container path {@link IPath} for standard VM "J2SE-1.5"
 	 */
+	@Deprecated(since = "2.30", forRemoval = true)
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JREContainerProvider.java:86]: @szarnekow ok this way?

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JREContainerProvider.java:86]: Absolutely

**Pull request merged**